src/crypto/mod.rs

```diff
@@ -0,0 +1,291 @@
+mod error;
+
+use byteorder::{BigEndian, ByteOrder};
+
+use pairing::{CurveAffine, CurveProjective, Engine, Field, PrimeField};
+use rand::{ChaChaRng, Rand, Rng, SeedableRng};
+use ring::digest;
+
+use self::error::{ErrorKind, Result};
+
+/// Returns a hash of the given message in `G2`.
+pub fn hash_g2<E, M>(msg: M) -> E::G2
+where
+    E: Engine,
+    <E as Engine>::G2: Rand,
+    M: AsRef<[u8]>,
+{
+    let digest = digest::digest(&digest::SHA256, msg.as_ref());
+    // The `pairing` crate's `G2` implements `Rand`. We initialize a seedable RNG with the SHA256
+    // digest, and use it to generate the element.
+    let mut msg_u32: Vec<u32> = Vec::with_capacity(256 / 32);
```

Comment:
Good idea! I changed it.
I went a bit overboard and used [init_with](https://docs.rs/init_with/1.1.0/init_with/) to avoid the `0u32` initialization. (But the bottleneck is certainly either `digest` or `from_seed`.)